src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1318,12 +1318,13 @@
                                            (assoc :event_hash (->> (shash/resource-event-identity-pkey %)
                                                                    (sutils/munge-hash-for-storage))))]
                    (when-not (empty? resource_events)
-                     (->> resource_events
-                          (sp/transform [sp/ALL :containment_path] #(some-> % sutils/to-jdbc-varchar-array))
-                          (map adjust-event)
-                          (map add-event-hash)
-                          (jdbc/insert-multi! :resource_events)
-                          dorun))
+                     (let [inserter (fn [x] (jdbc/insert-multi! :resource_events x {:on-conflict "do nothing"}))]
+                       (->> resource_events
+                            (sp/transform [sp/ALL :containment_path] #(some-> % sutils/to-jdbc-varchar-array))
+                            (map adjust-event)
+                            (map add-event-hash)
+                            (inserter)
+                            dorun)))
```

Comment:
I might be tempted to call this `insert!` or something, rather than `inserter` and it's typical to leave off the parens in a threading form when there are no args, but neither is important.